concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/ConcordiumHdWallet.java

```diff
@@ -0,0 +1,328 @@
+        MnemonicCode mnemonicCode = new MnemonicCode(in, BIP39_ENGLISH_SHA256);
+
+        // This throws an exception if the seed phrase is not valid.
+        mnemonicCode.check(seedPhrase);
+
+        byte[] seed = MnemonicCode.toSeed(seedPhrase, "");
+        StringBuilder sb = new StringBuilder();
+        for (byte b : seed) {
+            sb.append(String.format("%02X ", b));
+        }
+        return new ConcordiumHdWallet(sb.toString(), network);
+    }
+
+    /**
+     * Create a Concordium hierarchical deterministic wallet from a seed.
+     * @param seedAsHex the seed encoded as a hex string. Must be 128 characters.
+     * @param network the network the wallet is for.
+     * @return a wallet that can derive required Concordium wallet keys
+     */
+    public static ConcordiumHdWallet fromHex(String seedAsHex, Network network) {
+        if (seedAsHex.length() != 128) {
+            throw new IllegalArgumentException(
+                    "The provided seed " + seedAsHex + " is invalid as its length was not 128.");
+        } else if (!HexadecimalValidator.isHexadecimal(seedAsHex)) {
+            throw new IllegalArgumentException("The provided seed " + seedAsHex + " is not a valid hexadecimal string");
+        }
+
+        return new ConcordiumHdWallet(seedAsHex, network);
+    }
+
+    private enum UnsignedIntegerType {
+        U8,
+        U32,
+        U64;
+    }
+
+    private void checkUnsignedIntegerValues(UnsignedIntegerType type, long ...values) {
+        long maxValue;
+        switch (type) {
+            case U8: 
+                maxValue = 255;
+                break;
+            case U32:
+                maxValue = 4294967295l;
+                break;
+            case U64:
+                // Note that a long does not allow for a full u64 as it is a i64.
+                // The types in the SDK would need to be updated to fully accomodate a u64,
+                // and at that point this should also be changed.
+                maxValue = Long.MAX_VALUE;
+                break;
+            default:
+                throw new IllegalArgumentException("Uknown unsigned integer type was provided");
+        }
+
+        for (long value : values) {
+            if (value > maxValue || value < 0) {
+                throw new IllegalArgumentException("The value " + value + " is not a valid " + type);
+            }
+        }
+    }
+
+    private interface ExtractKey {
+        String getKey(String seedAsHex, String network);
+    }
+    
+    private String getKeyResult(ExtractKey extractor) {
+        KeyResult result = null;
+        try {
+            String jsonStr = extractor.getKey(this.seedAsHex, this.network.getValue());
+            result = JsonMapper.INSTANCE.readValue(jsonStr, KeyResult.class);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+        if (!result.isSuccess()) {
```

Comment:
Shouldn't this be checked before trying to deserialize the json?

Reply:
Without deserializing I would not be able to inspect the field. What I have done here is basically a copy-paste of the approach used elsewhere in the SDK. You can search for `isSuccess()` to see how it is elsewhere as well.